src/pages/reports/index/Reports.tsx

```diff
@@ -324,9 +341,21 @@ export function Reports() {
     setIsPendingExport(true);
     setErrors(undefined);
 
-    request('POST', endpoint(report.endpoint), report.payload, {
-      responseType: report.payload.send_email ? 'json' : 'blob',
-    })
+    const { client_id } = report.payload;
+
+    request(
+      'POST',
+      endpoint(report.endpoint),
+      {
+        ...report.payload,
+        ...(report.identifier === 'product_sales' && {
+          client_id: !client_id ? null : client_id,
+        }),
+      },
```

Comment:
Can we move this level above with a clear `if`? This way it's a bit unreadable and I can imagine someone crying trying to read this one :laughing:

Reply:
😄 , The point of those conditions is to not send `client_id` property to the API if it is not needed. I believe that I made more clear code now. Changed!